src/pages/resultsView/SurvivalStoreHelper.ts

```diff
@@ -0,0 +1,29 @@
+import {Patient, ClinicalData} from "shared/api/generated/CBioPortalAPI";
+import {PatientSurvival} from "../../shared/model/PatientSurvival";
+
+export function getPatientSurvivals(survivalClinicalDataGroupByUniquePatientKey: any, patients: Patient[], 
+    targetUniquePatientKeys: string[], statusAttributeId: string, monthsAttributeId: string, 
+    statusFilter: (s: string) => boolean): PatientSurvival[] {
+
+    let patientSurvivals: PatientSurvival[] = [];
+    if (targetUniquePatientKeys) {
+        targetUniquePatientKeys.forEach((uniquePatientKey: string) => {
+            const clinicalData: ClinicalData[] = survivalClinicalDataGroupByUniquePatientKey![uniquePatientKey];
+            if (clinicalData) {
+                const statusClinicalData: ClinicalData = clinicalData.filter(c => c.clinicalAttributeId === statusAttributeId)[0];
```

Comment:
you could use [].find instead of [].filter here.  It will return first found item or undefined. Will stop iteration when it finds item and also doesn't require index.   You should double check though that it works in IE11 via babel-polyfill.